[Zhao Yao] iP

WingKei67 opened the pull request:
# YiliaPro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

# 
If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
Customized features

Commit:
Added list

Commit:
Added mark and unmark function

Commit:
Extends task

Commit:
Added testing cases

Commit:
Handled exceptions

Commit:
Added delete

Commit:
Added enum

Commit:
Enabled file saving

Commit:
Added localdate

Commit:
Merged branch-level-7

Commit:
Merged branch-level-8

Commit:
Add more OOP

Commit:
Add packages

Commit:
Merge remote-tracking branch 'upstream/add-gradle-support'

Add the Gradle wrapper files to the project

Commit:
Add Gradle

Commit:
Add unit testing

Commit:
Add JAR file

Commit:
Modify README.md

Commit:
Add JavaDoc

Commit:
Modify coding standard

Commit:
Add find command

Commit:
Add find command

Commit:
Check coding styles

Commit:
Add a GUI

Commit:
Modify Varargs

Commit:
Merge branch 'branch-level-10'

Add GUI to current branch

BlopApple commented on src/main/java/yilia/MainWindow.java:
```diff
@@ -0,0 +1,51 @@
+    private void handleUserInput() {
+        String input = userInput.getText();
+        String response = yilia.getResponse(input);
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(input, userImage),
+                DialogBox.getYiliaDialog(response, yiliaImage)
+        );
```
I think the closing parentheses on line 48 should not be on a separate line.

BlopApple commented on src/main/java/yilia/Yilia.java:
```diff
@@ -0,0 +1,141 @@
+    private void handleUserInput() {
+        String userText = userInput.getText();
+        String yiliaText = getResponse(userInput.getText());
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(userText, user),
+                DialogBox.getYiliaDialog(yiliaText, yilia)
+        );
```
I think the closing parentheses on line 103 should not be on a separate line

BlopApple reviewed (commented):
I think your code is pretty well written in terms of coding standards. Also interesting name for your chat bot. Overall I think your code is quite well done. Keep it up!

Commit:
Fix some bugs

minjiunn commented on src/main/java/yilia/Parser.java:
```diff
@@ -0,0 +1,89 @@
+     * @throws YiliaException  If the command is unknown to Yilia.
+     * @throws NoIndexException  If the index is not given by users.
+     */
+    public static Command parse(String text) throws DescriptionEmptyException, KeywordMissingException,
```
i feel like adding too many if-else statements here might make the code a little cluttered and affect readability, just something to consider!

minjiunn commented on src/main/java/yilia/Type.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Represents four different types of a task.
+ */
+public enum Type {
```
maybe the name for the enum could be more specific? eg TypeOfTask

minjiunn commented on src/main/java/yilia/exception/YiliaException.java:
```diff
@@ -0,0 +1,16 @@
+package yilia.exception;
```
you have many exception classes, maybe you can consider letting them inherit from one exception parent class.

minjiunn reviewed (commented):
good job overall, i like that no coding conventions were seriously violated :)

Commit:
Revert "Fix some bugs"

This reverts commit ebd725473aec78a6f9d7fda2f1c8408f1a053551.

Revert to last week's progress.

Commit:
Add assertions

Commit:
Improve code quality

Commit:
Find Command: complete find feature using streams.
TaskList: enable a task list to be a stream.

Find Command can be easier done using streams.

Using streams feature can complete more functionalities and improve
efficiency.

WingKei67 committed:
Merge pull request #2 from WingKei67/branch-A-Assertions

Add assertions

WingKei67 committed:
Merge pull request #3 from WingKei67/branch-A-CodeQuality

Improve code quality

WingKei67 committed:
Merge branch 'master' into branch-A-Streams

WingKei67 committed:
Merge pull request #4 from WingKei67/branch-A-Streams

Find Command: complete find feature using streams.

Commit:
Use GitHub Actions to set up Continuous Integration (CI).

GitHub Actions is CI/CD tool integrated into GitHub.

GitHub Actions will run the workflow (as per the .yml file) every time certain project
events are triggered (e.g., when a PR is updated, or the master branch is updated).

Commit:
Modify gradle file

Commit:
Fix bugs

Commit:
Add Type file

Commit:
Change gradle file

Commit:
Add a new line

Commit:
Improve UIs

Commit:
Add personalities

Commit:
Add more exception handlers

Commit:
Add more unit testing

Commit:
Add userguide

Commit:
Add jar

Commit:
Finalize the product

Commit:
Fix some bugs
